docs/linux/external_fuzzing_usb.md

```diff
@@ -136,36 +136,21 @@ These instructions describe how to set this up on a Raspberry Pi Zero W, but any
 9. Download syzkaller, apply the patch below and build `syz-executor`:
 
 ``` c
-diff --git a/executor/common_usb.h b/executor/common_usb.h
-index e342d808..278c2f4e 100644
---- a/executor/common_usb.h
-+++ b/executor/common_usb.h
-@@ -269,9 +269,7 @@ static volatile long syz_usb_connect(volatile long a0, volatile long a1, volatil
-
-    // TODO: consider creating two dummy_udc's per proc to increace the chance of
-    // triggering interaction between multiple USB devices within the same program.
+diff --git a/executor/common_usb_linux.h b/executor/common_usb_linux.h
+index 451b2a7b..64af45c7 100644
+--- a/executor/common_usb_linux.h
++++ b/executor/common_usb_linux.h
+@@ -292,9 +292,7 @@ static volatile long syz_usb_connect_impl(uint64 speed, uint64 dev_len, const ch
+
+        // TODO: consider creating two dummy_udc's per proc to increace the chance of
+        // triggering interaction between multiple USB devices within the same program.
 -       char device[32];
 -       sprintf(&device[0], "dummy_udc.%llu", procid);
--       rv = usb_raw_init(fd, speed, "dummy_udc", &device[0]);
+-       int rv = usb_raw_init(fd, speed, "dummy_udc", &device[0]);
 +       rv = usb_raw_init(fd, speed, "20980000.usb", "20980000.usb");
-    if (rv < 0) {
-            debug("syz_usb_connect: usb_raw_init failed with %d\n", rv);
-            return rv;
-diff --git a/executor/executor.cc b/executor/executor.cc
-index 34949a01..1afcb288 100644
---- a/executor/executor.cc
-+++ b/executor/executor.cc
-@@ -604,8 +604,8 @@ retry:
-                    call_extra_cover = true;
-            }
-            if (strncmp(syscalls[call_num].name, "syz_usb_connect", strlen("syz_usb_connect")) == 0) {
--                       prog_extra_timeout = 2000;
--                       call_extra_timeout = 2000;
-+                       prog_extra_timeout = 5000;
-+                       call_extra_timeout = 5000;
-            }
-            if (strncmp(syscalls[call_num].name, "syz_usb_control_io", strlen("syz_usb_control_io")) == 0)
-                    call_extra_timeout = 300;
+        if (rv < 0) {
```

Comment:
Is this still needed? I thought we up-streamed all necessary bits. If this is not needed, then let's drop this step.
If this is needed then let's upstream this and drop this step.
@xairy

Reply:
We upstreamed everything for fuzzing in VMs. These instructions are for running reproducers with a Raspberry Pi Zero. Proper upstream support for that along with fuzzing is on the USB TODO list, but this will require some effort.